satellite/marketing/marketingweb/server.go

```diff
@@ -32,65 +32,89 @@ type Server struct {
 
 	listener net.Listener
 	server   http.Server
+
+	templateDir string
+}
+
+func (s *Server) templateFilePath(name string) string {
+	return filepath.Join(s.config.StaticDir, "pages", name)
 }
 
-// The three pages contained in addPages are pages all templates require
-// This exists in order to limit handler verbosity
-func (s *Server) addPages(assets []string) []string {
-	rp := s.config.StaticDir + "/pages/"
-	pages := []string{rp + "base.html", rp + "index.html", rp + "banner.html"}
-	for _, page := range assets {
-		pages = append(pages, page)
+// commonPages returns templates that are required for everything.
+func (s *Server) commonPages() []string {
+	return []string{
+		filepath.Join(s.templateDir, "base.html"),
+		filepath.Join(s.templateDir, "index.html"),
+		filepath.Join(s.templateDir, "banner.html"),
 	}
-	return pages
 }
 
 // NewServer creates new instance of offersweb server
 func NewServer(logger *zap.Logger, config Config, listener net.Listener) *Server {
-	server := Server{
+	s := &Server{
 		log:      logger,
 		config:   config,
 		listener: listener,
 	}
 
-	logger.Sugar().Debugf("Starting Marketing Admin UI on %s...", server.listener.Addr().String())
-	fs := http.FileServer(http.Dir(server.config.StaticDir))
-	mux := mux.NewRouter()
-	if server.config.StaticDir != "" {
+	logger.Sugar().Debugf("Starting Marketing Admin UI on %s...", s.listener.Addr().String())
+
+	fs := http.FileServer(http.Dir(s.config.StaticDir))
+	mux := http.NewServeMux()
+	if s.config.StaticDir != "" {
 		mux.Handle("/static/", http.StripPrefix("/static", fs))
-		mux.Handle("/", http.HandlerFunc(server.appHandler))
-	}
-	server.server = http.Server{
-		Handler: mux,
+		mux.Handle("/", s)
 	}
+	s.server.Handler = mux
 
-	return &server
+	s.templateDir = filepath.Join(s.config.StaticDir, "pages")
+
+	return s
 }
 
-// appHandler is web app http handler function
-func (s *Server) appHandler(w http.ResponseWriter, req *http.Request) {
+// ServeHTTP handles index request
+func (s *Server) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	if req.URL.Path != "/" {
-		s.serveError(w, req)
+		s.serveInternalError(w, req)
```

Comment:
Why the correct one is an internal server error when the path doesn't match the ones exposed?

Reply:
Ah, right... yeah, my mistake.